src/sys/linux/hardwareinfo.rs

```diff
@@ -0,0 +1,38 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::fs::File;
+use std::io::Read;
+use ::sys::linux::helpers;
+
+impl ::HardwareInfo for ::HostInfo {
+    fn online_cpu_count(&self) -> u32 {
+        let mut file = File::open("/sys/devices/system/cpu/online").unwrap();
```

Comment:
I discussed this with @waywardmonkeys as well. We should probably handle all the `unwrap()` scenarios carefully without `panic`'ing. I feel like returning a "safe" value (like 1 core, 0 memory, etc) or `Result<>`'s or `Option<>`'s might be a good idea. But that will require the API and other platforms to be modified as well.

However, not sure we need to add support for older kernels, and we can't really check for newer kernel versions (we don't know when sysfs will be refactored, if ever). This part of sysfs has been stable and documented [since at least 2008](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=d62720ade82c5e5b8f9585e5ed02c89573ebf111).